### ISSUE TITLE
Question about changing image size

### ISSUE THREAD

**Issue by KKaiYH** [XuJiacong/PIDNet#72]
The problem I encountered is that I changed the train image size 512 in the yaml file to train model, and I also changed the test image size, but when I actually ran the testing, the image size of the test didn't change according to the settings in the yaml file, so I would like to ask if there is any error in this part of the content ?
As shown below, because the testing image size is defined in multi_scale_aug in the resize part, but multi_scale=False in test_dataset, so it doesn't do resize at all ?
![Q1](https://github.com/XuJiacong/PIDNet/assets/134195214/64bdd007-15ed-447c-8ae4-35057831ba9e)
![Q2](https://github.com/XuJiacong/PIDNet/assets/134195214/e70a30d9-0d6b-456f-a32a-eda8c3084130)

**Comment by lpetflo** [comment]
That's correct, PIDNet only performs resizing if multi_scale is True. However I personally wouldn't call this in issue as I want to test my predictions on full size whatsoever. I'm currently training on a reduced scale, but am guessing that the model is correctly generalizing to the original size due to the training with multiple scales. 
If you really wanted to test on downscaled versions of your dataset you would have to change the code.

**Comment by KKaiYH** [comment]
Thank you for your reply !
I would like to ask about the training you tried on the reduced size, did
you try to test it on the same reduced size ?
I'm currently having problems with a significant drop in accuracy if I test
my predictions on the reduced size, providing some of my experimental data
in hopes of further discussion with you.
If you have any related experiments, hope you can share them with me.
Thank you !
[image: image.png]
Florian S. ***@***.***> 於 2023年11月2日 週四 下午5:04寫道：

**Comment by lpetflo** [comment]
Yes, I actually did some testing on 50% prediction as well as using upscaling for full-scale predictions.
Using PIDNet and my highly imbalanced binary dataset, I was able to get 71.92% mIoU when evaluating the 50% downscaled predictions on the training I did at that 50% size (using the multi-scale option however). Using the authors' native full-scale approach, I could only get 70.48%, which was in line with my expectations. Using [TDNet](https://github.com/feinanshan/TDNet) as a segmentation model (which doesn't use native full-scale predictions), I got the same 1-2% degradation when using full-scale predictions instead of the 50% predictions I trained it on.
Due to my limited ressources I couldn't try full-scale training so your findings are still quite interesting for me!
